### PR TITLE
Meta: Remove unused macros from all_the_debug_macros.cmake

### PR DESCRIPTION
**Pull request by BenWiederhake**
Looks like a few macros are no longer used, or were copied instead of being renamed.

Commands that were helpful while investigating this:
```
$ grep -P '^set' Meta/CMake/all_the_debug_macros.cmake | sed -Ee 's,set\((.+) ON\)$,\1,' > macros.lst
$ for i in $(cat macros.lst); do echo -n "$i "; git grep -Pn '\b'"$i"'\b' | wc -l; done | tee matches.lst
$ sort -k2 -n matches.lst
```

<details><summary>Fun statistics</summary>

You'll have to subtract 3 from these numbers, as each macros is mentioned twice in `AK/Debug.h.in` and once in ``

The least-often used debug macros are (note that these are actually only used once or twice in the C++ code):
- COMMIT_DEBUG 4
- COPY_DEBUG 4
- DIFF_DEBUG 4
- DISASM_DUMP_DEBUG 4
- DUMP_REGIONS_ON_CRASH 4
- ESCAPE_SEQUENCE_DEBUG 4
- FUTEX_DEBUG 4
- GPT_DEBUG 4
- HIGHLIGHT_FOCUSED_FRAME_DEBUG 4
- HPET_COMPARATOR_DEBUG 4
- HUNKS_DEBUG 4
- ICMP_DEBUG 4
- IMAGE_LOADER_DEBUG 4
- ITEM_RECTS_DEBUG 4
- KEYBOARD_DEBUG 4
- MARKDOWN_DEBUG 4
- MBR_DEBUG 4
- MENU_DEBUG 4
- MULTIPROCESSOR_DEBUG 4
- NETWORK_TASK_DEBUG 4
- POLL_SELECT_DEBUG 4
- PROCFS_DEBUG 4
- SCHEDULER_RUNNABLE_DEBUG 4
- SHELL_JOB_DEBUG 4
- SLAVEPTY_DEBUG 4
- SOLITAIRE_DEBUG 4
- SYSTEM_MENU_DEBUG 4
- SYSTEMSERVER_DEBUG 4
- UPDATE_COALESCING_DEBUG 4
- VIRTUAL_CONSOLE_DEBUG 4
- WASI_DEBUG 4
- WINDOWMANAGER_DEBUG 4
- CALLBACK_MACHINE_DEBUG 5
- CPP_DEBUG 5
- ELF_IMAGE_DEBUG 5
- EMOJI_DEBUG 5
- ETHERNET_DEBUG 5
- EXEC_DEBUG 5
- FILE_CONTENT_DEBUG 5
- FILL_PATH_DEBUG 5
- GEMINI_DEBUG 5
- GHASH_PROCESS_DEBUG 5
- HTTPJOB_DEBUG 5
- IPV4_DEBUG 5
- LOCK_RANK_ENFORCEMENT 5
- LOCK_RESTORE_DEBUG 5
- MASTERPTY_DEBUG 5
- MEMORY_DEVICE_DEBUG 5
- NETWORKJOB_DEBUG 5
- NT_DEBUG 5
- PATH_DEBUG 5
- PTMX_DEBUG 5
- SERVICE_DEBUG 5
- SHARED_QUEUE_DEBUG 5
- STORAGE_DEVICE_DEBUG 5
- SYSFS_DEBUG 5
- UDP_DEBUG 5
- WSMESSAGELOOP_DEBUG 5

And the most-often used debug macros are:
- OPENTYPE_GPOS_DEBUG 50
- TLS_DEBUG 72
- WEBDRIVER_DEBUG 73
- CSS_PARSER_DEBUG 76
- SQL_DEBUG 85

### CODE REVIEW

File: Meta/refresh-serenity-qtcreator.sh [hunk 1]
```diff
@@ -11,4 +11,4 @@ fi
 
 cd "$SERENITY_SOURCE_DIR"
 
-find . \( -name Base -o -name Patches -o -name Ports -o -name Root -o -name Toolchain -o -name Build \) -prune -o \( -name '*.ipc' -or -name '*.cpp' -or -name '*.idl' -or -name '*.c' -or -name '*.h' -or -name '*.S' -or -name '*.css' -or -name '*.cmake' -or -name '*.json' -or -name '*.gml' -or -name 'CMakeLists.txt' \) -print > serenity.files
+find . \( -name Base -o -name Patches -o -name Ports -o -name Root -o -name Toolchain -o -name Build \) -prune -o \( -name '*.ipc' -or -name '*.cpp' -or -name '*.idl' -or -name '*.c' -or -name '*.h' -or -name '*.in' -or -name '*.S' -or -name '*.css' -or -name '*.cmake' -or -name '*.json' -or -name '*.gml' -or -name 'CMakeLists.txt' \) -print > serenity.files
```
Comment (ADKaster) [inline Meta/refresh-serenity-qtcreator.sh:14]: These changes seem fine, but ... can QtCreator understand `compile_commands.json` yet? 😅 . Maybe there's something else about our CMake setup that makes us have to write this massive files list though.

**Review by ADKaster — APPROVED** [review]
Thanks for cleaning this up! Love to see a red PR (on the changes, not on the CI :P)

File: Meta/refresh-serenity-qtcreator.sh (hunk 1 above)
Comment (BenWiederhake) [reply Meta/refresh-serenity-qtcreator.sh:14]: I'm not sure how that could be done, if you have suggestions I'd be happy to hear them ^^

Comment (BenWiederhake) [reply Meta/refresh-serenity-qtcreator.sh:14]: All that I can find is a [six-year old dead forum thread](https://forum.qt.io/topic/79805/qtcreator-clang-code-model-with-compile_commands-json) where it was suggested and (apparently) never followed-up.

**Pull request merged**